web/src/peer_data.ts

```diff
@@ -453,3 +471,40 @@ export async function get_unique_subscriber_count_for_streams(
     }
     return valid_subscribers.size;
 }
+
+export async function load_subscriptions_for_user(user_id: number): Promise<void> {
+    if (has_complete_subscriber_data()) {
+        return;
+    }
+
+    if (pending_subscription_requests.has(user_id)) {
+        return pending_subscription_requests.get(user_id)!;
+    }
```

Comment:
Oh, this seems to make no effort to keep track of whether we've fetched this specific user's subscription data previously? Given we have live updates, that seems like it'd be useful to do? 

(And the server can probably be tweaked to tell us which ones we start with complete data for, so it'd be valuable for the web app to have a field for tracking that)

Reply:
Sure, that's easy to add, added `fetched_user_subscriptions`.

The server tweak seems like something for a future PR, but yeah that would also be good to do eventually.